lib/s3/managed_upload.js

```diff
@@ -215,7 +239,7 @@ AWS.S3.ManagedUpload = AWS.util.inherit({
   validateBody: function validateBody() {
     var self = this;
     self.body = self.service.config.params.Body;
-    if (!self.body) throw new Error('params.Body is required');
+    if (self.body === undefined) throw new Error('params.Body is required');
```

Comment:
Can you confirm this won't cause https://github.com/aws/aws-sdk-js/issues/570 to happen again? Looks like this change was made in the past, and caused a regression. Just want to make sure we don't cause the same regression again.